src/LiveComponent/src/ComponentWithFormTrait.php

```diff
@@ -288,7 +288,7 @@ private function extractFormValues(FormView $formView): array
                 && !$child->vars['multiple']                // is not multiple
                 && !\is_string($child->vars['placeholder'])  // has no placeholder (empty string is valid)
             ) {
-                $choices = $child->vars['choices'];
+                $choices = $child->vars['preferred_choices'] ?: $child->vars['choices'];
```

Comment:
Could you add some comment here (more or less as the other lines) to keep this always-expanding list ? 

(i'm thinking to the next ones that would need to understand the whole system with no comments 😅 )